hs_discover/views.py

```diff
@@ -1,57 +1,224 @@
+                    # (datefilter.start < start_date < datefilter.end) or (start_date < datefilter.start)
+                    sqs = sqs.exclude(start_date__gt=datefilter.end).exclude(end_date__lt=datefilter.start)
+                except ValueError as e:
+                    pass  # ignore bad values and don't filter which is the correct action
+        except TypeError as type_ex:
+            pass  # no filters passed "the JSON object must be str, bytes or bytearray not NoneType"
 
-        vocab = [x for x in vocab if len(x) > 2]
-        vocab = list(set(vocab))
-        vocab = sorted(vocab)
+        except json.JSONDecodeError as parse_ex:
+            return JsonResponse({'message': 'Filter JSON parsing error - {}'.format(str(parse_ex)),
+                                 'received': request.query_params}, status=400)
 
-        if q:
-            sqs = sqs.filter(content=q).boost('keyword', 2.0)
+        except Exception as gen_ex:
+            logger.debug('hs_discover API - {}: {}'.format(type(gen_ex), str(gen_ex)))
+            return JsonResponse({'message': '{}'.format('{}: query error. Contact a server administrator.'
+                                                        .format(type(gen_ex)))}, status=520)
+
+        sqs = sqs.order_by(sort)
 
         resources = []
-        for result in sqs:
+
+        p = Paginator(sqs, self.perpage)
+
+        pnum = 1
+        if request.GET.get('pnum'):
+            pnum = request.GET.get('pnum')
+            pnum = max(1, int(pnum))
+            pnum = min(pnum, p.num_pages)
+
+        geodata = []
+
+        for result in p.page(pnum):
+            contributor = 'None'  # contributor is actually a list and can have multiple values
+            owner = 'None'  # owner is actually a list and can have multiple values
+            author_link = None  # Send None to avoid anchor render
+            creator = 'None'
+
+            if result.creator:
+                creator = result.creator
+
+            if result.author:
+                author_link = result.author_url
+                author = str(result.author)
+            else:
+                if result.authors:
+                    author = result.authors[0]
+                elif result.organization:
+                    author = result.organization
+                else:
+                    author = 'None'
+
+            if result.contributor is not None:
+                try:
+                    contributor = result.contributor
+                except:
+                    pass
+
+            if result.owner is not None:
+                try:
+                    owner = result.owner
+                except:
+                    pass
+            pt = ''  # pass empty string for the frontend to ensure the attribute exists but can be evaluated for empty
+            try:
+                if 'box' in result.coverage_type:
+                    pt = {'short_id': result.short_id, 'title': result.title, 'coverage_type': 'region'}
```

Comment:
I'm curious why you're mapping box to region here.  You're wrote the use of this parameter... I don't think we should be mapping to a new term if we don't have to.

Reply:
Box is what is used in Resource edit, going back to Box. Originally was going to consider updating help docs for "Region". Undoing. Commit ceeb0d604